src/main/java/com/artipie/RepoPermissions.java

```diff
@@ -110,8 +118,37 @@ public FromSettings(final Settings settings) {
         }
 
         @Override
-        public CompletionStage<Map<String, List<String>>> get(final String repo) {
-            throw new UnsupportedOperationException("To implemented later");
+        public CompletionStage<Map<String, List<String>>> permissions(final String repo) {
+            return this.yaml(new Key.From(String.format("%s.yaml", repo))).thenApply(
+                yaml -> Optional.ofNullable(yaml.yamlMapping("repo").yamlMapping("permissions"))
+                .map(
+                    perms -> {
+                        final Map<String, List<String>> res = new HashMap<>();
+                        perms.keys().forEach(
+                            node -> res.put(
+                                node.asScalar().value(),
+                                perms.yamlSequence(node.asScalar().value()).values().stream()
+                                .map(item -> Scalar.class.cast(item).value())
```

Comment:
@olenagerasimova dynamic casting via `Scalar.class.cast(item)` could be avoided by using `item.asScalar()` method